### PR TITLE
Refactor function tests and disable custom function auth tests

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Tests using custom function auth are disabled. In order to continue to test Altas functions, the tests have been split into multiple tests.

### CODE REVIEW

**Review by elle-j — APPROVED** [review]

File: integration-tests/tests/src/tests/sync/user.ts [hunk 1]
```diff
@@ -530,7 +522,7 @@ describe.skipIf(environment.missingServer, "User", () => {
             const { username, secret } = loginPayload;
 
             if (secret !== "v3ry-s3cret") {
-              throw new Error("Ah ah ah, you didn't say the magic word");
+              throw new Error("Ah ah ah, you did not say the magic word");
```
Comment (takameyer) [inline integration-tests/tests/src/tests/sync/user.ts:525]: Probably can revert this

File: integration-tests/tests/src/tests/sync/user.ts [hunk 2]
```diff
@@ -548,12 +540,17 @@ describe.skipIf(environment.missingServer, "User", () => {
...
         .function({
-          can_evaluate: {},
-          name: "sumFunc",
+          name: "resetFunc",
           private: false,
+          can_evaluate: {},
           source: `
-            exports = function (...args) {
-              return parseInt(args.reduce((a, b) => a + b, 0));
+            exports = ({ token, tokenId, username, password }) => {
```
Comment (takameyer) [inline integration-tests/tests/src/tests/sync/user.ts:547]: Intentional for this PR?

File: integration-tests/tests/src/tests/sync/user.ts [hunk 3]
```diff
@@ -585,6 +582,23 @@ describe.skipIf(environment.missingServer, "User", () => {
 
       await user.logOut();
     });
+  });
+
+  describe("custom functions", () => {
+    importAppBefore(
+      buildAppConfig("with-custom-function")
```
Comment (takameyer) [inline integration-tests/tests/src/tests/sync/user.ts:589]: Was this also intentional?

File: integration-tests/tests/src/tests/credentials/function.ts [hunk 4]
```diff
@@ -33,7 +36,7 @@ describe.skipIf(environment.missingServer, "custom-function credentials", () =>
           const { username, secret } = loginPayload;
 
           if (secret !== "v3ry-s3cret") {
-            throw new Error("Ah ah ah, you didn't say the magic word");
+            throw new Error("Ah ah ah, you did not say the magic word");
```
Comment (takameyer) [inline integration-tests/tests/src/tests/credentials/function.ts:39]: Could also revert this

**Review by takameyer — APPROVED** [review]
LGTM
Hopefully we can figure this out later.

File: integration-tests/tests/src/tests/credentials/function.ts (hunk 4 above)
Comment (kneth) [reply integration-tests/tests/src/tests/credentials/function.ts:39]: true

File: integration-tests/tests/src/tests/sync/user.ts (hunk 2 above)
Comment (kneth) [reply integration-tests/tests/src/tests/sync/user.ts:547]: Yes, the `sumFunc` is moved to line 590

**Pull request merged**